### PR TITLE
Improve an average late time for retry.

### PR DESCRIPTION
**Pull request by MtzwOsk**
I assume that less blocking code is much better :), so retry latency should be smaller than 1 second. 
1 second for request is over estimated. We can for that reason assume average time depends on max request in 1 minute. 
Maybe much better and precisely result could be compare delta between two oldest timestamp.

### CODE REVIEW

**Comment by MtzwOsk** [comment]
I have problem with sign license by GitHub account. I have signed, but status in still pending.

File: shipbob/client.py [hunk 1]
```diff
@@ -22,6 +22,10 @@ class ShipBobClient(Session):
 
     MAX_PAGE_SIZE: int = 250
     MAX_REQUESTS_PER_MINUTE: int = 150  # As per ShipBob docs, 150 reqs/min sliding window
+    LIMIT_PERIOD_FOR_REQUESTS_MAX_IN_SECONDS = 60
+    AVERAGE_DELAY_FOR_RETRY_IN_SECOND = float(
```
Comment (lullis) [inline shipbob/client.py:26]: This is a very convoluted way of just defining a float.

File: shipbob/client.py [hunk 2]
```diff
@@ -36,14 +40,14 @@ def __init__(self):
...
-        one_minute_old = now - timedelta(seconds=60)
+        one_minute_old = now - timedelta(seconds=self.MAX_REQUESTS_PER_MINUTE)
         self._request_bucket = [rt for rt in self._request_bucket if rt >= one_minute_old]
 
     def request(self, method, url, retry_on_rate_limit=True, *args, **kwargs):
         while len(self._request_bucket) >= self.MAX_REQUESTS_PER_MINUTE:
             logger.warning("Close to rate limit. Sleeping for one second")
             self._clear_request_bucket()
-            time.sleep(1)
+            time.sleep(self.AVERAGE_DELAY_FOR_RETRY_IN_SECOND)
```
Comment (lullis) [inline shipbob/client.py:50]: What average is this, if it is always going to result in `60 / 150` (0.4) seconds of sleep time?

**Comment by lullis** [comment]
Hello @MtzwOsk, and apologies for the delayed response.

It seems like you used a different email address than the one from your github account, this is why it is showing an error.

Thank you for the PR, but I don't see how your solution is an improvement over the existing implementation. Shipbob clearly states that they have a *sliding window* of 150 requests, so I don't see the benefit of making that calculation.

File: shipbob/client.py (hunk 1 above)
Comment (MtzwOsk) [reply shipbob/client.py:26]: True, normal    divided is fine

 AVERAGE_DELAY_FOR_RETRY_IN_SECOND = LIMIT_PERIOD_FOR_REQUESTS_MAX_IN_SECONDS / MAX_REQUESTS_PER_MINUTE

File: shipbob/client.py (hunk 2 above)
Comment (MtzwOsk) [reply shipbob/client.py:50]: Thank You for your answer. 

Ofc, now is 0.4 second, but its 2 times less time waiting for service.  
Just imagine, that you use other libraries with I/O, and here we explicite stop working on 1 sec. I don't see any reason for 1 second, could you tell me why exactly 1 second ? 
My average  is not a ideal resolve, but it closer to life.  
I hope that request duration for that API is not 1 second. :) 

Second Intension, but I haven't did add yet was get to user possible to set other value for MAX_REQUESTS_PER_MINUTE. I see one reason why, we don't know the future, who knows, maybe API max request tomorrow will be change. When ShipBob API change that value we are force to update our code and force user to update our package. 
Just imagine, MAX_REQUESTS_PER_MINUTE = 60 becouse you dont have any subscribe or somthing simallar. 
Now tell to user from all the world to update package.
Flexible is worth.

**Pull request closed without merging**